atlasdb-client/src/test/java/com/palantir/atlasdb/transaction/impl/ReadTransactionShould.java

```diff
@@ -0,0 +1,165 @@
+            TableReference.createWithEmptyNamespace("dummy-conservative");
+    private static final TableReference DUMMY_THOROUGH_TABLE =
+            TableReference.createWithEmptyNamespace("dummy-thorough");
+    private static final Cell DUMMY_CELL = Cell.create("row".getBytes(), "col".getBytes());
+    private static final byte[] EMPTY_BYTES = "".getBytes();
+
+    private static final Map<String, Object[]> simpleGets = ImmutableMap.<String, Object[]>builder()
+            .put("get", new Object[] {DUMMY_THOROUGH_TABLE, ImmutableSet.of(DUMMY_CELL)})
+            .put("getRows", new Object[] {DUMMY_THOROUGH_TABLE, ImmutableList.of(EMPTY_BYTES), ColumnSelection.all()})
+            .put("getRange", new Object[] {DUMMY_THOROUGH_TABLE, RangeRequest.all()})
+            .put("getRanges", new Object[] {DUMMY_THOROUGH_TABLE, ImmutableList.of(RangeRequest.all())})
+            .build();
+
+    private ReadTransaction readTransaction;
+    private AbstractTransaction delegateTransaction;
+
+    @Before
+    public void setUp() throws Exception {
+        delegateTransaction = Mockito.mock(AbstractTransaction.class);
+        SweepStrategyManager sweepStrategies = Mockito.mock(SweepStrategyManager.class);
+        when(sweepStrategies.get()).thenReturn(ImmutableMap.of(
+                DUMMY_CONSERVATIVE_TABLE,
+                TableMetadataPersistence.SweepStrategy.CONSERVATIVE,
+                DUMMY_THOROUGH_TABLE,
+                TableMetadataPersistence.SweepStrategy.THOROUGH));
+        readTransaction = new ReadTransaction(delegateTransaction, sweepStrategies);
+    }
+
+    @Test
+    public void notAllowPuts() {
+        checkThrowsAndNoInteraction(() -> readTransaction.put(
+                DUMMY_CONSERVATIVE_TABLE,
+                ImmutableMap.of(DUMMY_CELL, "value".getBytes())),
+                IllegalArgumentException.class,
+                "is a read only transaction");
+    }
+
+    @Test
+    public void notAllowDeletes() {
+        checkThrowsAndNoInteraction(() -> readTransaction.delete(
+                DUMMY_CONSERVATIVE_TABLE,
+                ImmutableSet.of(DUMMY_CELL)),
+                IllegalArgumentException.class,
+                "is a read only transaction");
+    }
+
+    @Test
+    public void allowGetsOnConservativeTables() {
+        ImmutableSet<Cell> cellToGet = ImmutableSet.of(DUMMY_CELL);
+        readTransaction.get(DUMMY_CONSERVATIVE_TABLE, cellToGet);
+        Mockito.verify(delegateTransaction, times(1)).get(eq(DUMMY_CONSERVATIVE_TABLE), eq(cellToGet));
+    }
+
+    @Test
+    public void notAllowGetsOnThoroughTables() {
+        ImmutableSet<Cell> cellToGet = ImmutableSet.of(DUMMY_CELL);
+        checkThrowsAndNoInteraction(() -> readTransaction.get(DUMMY_THOROUGH_TABLE, cellToGet),
+                IllegalStateException.class,
+                "Cannot read");
+    }
+
+    @Test
+    public void notAllowSimpleGets() throws IllegalAccessException {
```

Comment:
It wasn't immediately clear to me that this is about _thorough_ simple gets. Suggest a rename to `notAllowSimpleGetsOnThoroughTables`.